lib/internal/util/inspect.js

```diff
@@ -312,7 +312,8 @@ function strEscape(str) {
 
   let result = '';
   let last = 0;
-  for (var i = 0; i < str.length; i++) {
+  let i = 0;
+  for (; i < str.length; i++) {
```

Comment:
How about initializing `i` inside for loop, and use the upper limit instead of `i` outside the loop?

For example in this case:
* line 316 becomes `for (let i = 0; i < str.length; i++) {`
* line 328 becomes `if (last !== str.length) {`

This way, we'll have one less variable to worry about.
Ditto at other places where iterator is initialized outside for loop

Reply:
Good optimization, will do that.

Reply:
Implemented. Also did another small optimization by pre-computing the last index in those loops. Should probably give a tiny bit of perf.